### ISSUE TITLE
[Sort] [Filter] Why is there no generic number parser that handle both integer and float?

### ISSUE THREAD

**Issue by JonasNo** [Mottie/tablesorter#841]
$.tablesorter.addParser({
    	id: 'float',
    	format: function(text) {
    		return parseFloat(text);
    	},
    	type: 'numeric'
    });

    $('#myTable').tablesorter({
    	debug: true,
    	widgets        : ['zebra', 'columns', 'filter', 'columnSelector'],
    	usNumberFormat : false,
    	sortReset      : true,
    	sortRestart    : true,
    	headers: {
    		1: {
    			sorter: 'float'
    		},
    		2: {
    			sorter: 'float'
    		},
    		4: {
    			sorter: 'percent'
    		}
    	}
    });

I have columns with both integer and float numbers.
Got the sorting to work but filtering only work partially. Then I tested "=111" it did not find "111.000". I have set the columns as sorter: 'float'.

My thinking is the parser should interpret the float as a number and match it.
Am I missing something here?

Example data:
With column values looking like:
89.000
111.000
130.000

Another example (mixed integer and float):
300
360
370
298.57
301.43
305.71
312.86

**Comment by Mottie** [comment]
Hi @JonasNo!

Try adding a `parsed: true` flag to the parser ([ref](http://mottie.github.io/tablesorter/docs/example-parsers.html)):

```js
$.tablesorter.addParser({
    id: 'float',
    format: function(text) {
        return parseFloat(text);
    },
    parsed : true, // filter widget flag
    type: 'numeric'
});
```

or you could add a [`filter-parsed` class name](http://mottie.github.io/tablesorter/docs/example-widget-filter.html#classes) to the header.

If that doesn't work, I'll set up a demo and test it further.

**Comment by Mottie** [comment]
Ok, I did end up looking at this in more depth.

First off, the default "digit" parser should work on this table: http://jsfiddle.net/Mottie/856bzzeL/72/

The reason it wasn't working with that original code is because:
* The [`usNumberFormat` option](http://mottie.github.io/tablesorter/docs/index.html#usnumberformat) is set to `false` so the internal format float function is looking for a non-U.S. number.
* `111.000` is parsed as `111000`.
* `111,000` would be parsed as `111`.

Also, to compare parsed values instead of actual values, you will still need to set the `filter-parsed` class on the column header. You will see this setting in the demo I shared.
